module-services/service-desktop/endpoints/reboot/RebootHelper.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2017-2023, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include <endpoints/Context.hpp>
+#include <endpoints/reboot/RebootHelper.hpp>
+#include <endpoints/JsonKeyNames.hpp>
+#include <endpoints/message/Sender.hpp>
+#include <service-desktop/DesktopMessages.hpp>
+#include <service-desktop/ServiceDesktop.hpp>
+
+#include <json11.hpp>
+
+namespace sdesktop::endpoints
+{
+    using sender::putToSendQueue;
+    auto RebootHelper::processPost(Context &context) -> ProcessResult
+    {
+        const auto rebootType = context.getBody()[json::reboot::rebootType].int_value();
+
+        switch (rebootType) {
+        case json::reboot::MSC:
+            sys::SystemManagerCommon::RebootMSC(owner);
+            return {sent::no, ResponseContext{.status = http::Code::Accepted}};
+        case json::reboot::reboot:
+            sys::SystemManagerCommon::Reboot(owner);
+            return {sent::no, ResponseContext{.status = http::Code::Accepted}};
+        case json::reboot::shutDown:
+            sys::SystemManagerCommon::RegularPowerDown(owner);
+            return {sent::no, ResponseContext{.status = http::Code::Accepted}};
+        }
```

Comment:
I would consider adding "default" in the end to make the code look better.

Reply:
I considered it, but I found this to be the most consistent with our codebase.